lib/upipe-modules/upipe_sync.c

```diff
@@ -1059,6 +1110,26 @@ static int upipe_sync_set_flow_def(struct upipe *upipe, struct uref *flow_def)
     return UBASE_ERR_NONE;
 }
 
+/** @internal @This sets the content of an sync option.
+ *
+ * @param upipe description structure of the pipe
+ * @param option name of the option
+ * @param content content of the option
+ * @return an error code
+ */
+static int upipe_sync_set_option(struct upipe *upipe,
+                                 const char *option, const char *content)
+{
+    struct upipe_sync *upipe_sync = upipe_sync_from_upipe(upipe);
+    assert(option != NULL);
```

Comment:
In most other pipes I've seen we usually check option and content and return UBASE_ERR_INVALID if either are NULL.